dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java

```diff
@@ -102,6 +102,22 @@ void testSetDataValuesFollowUp_Multi()
         assertFollowups( false, true, false );
     }
 
+    @Test
+    public void testAddDataValueWithBody()
+    {
+        String body = String.format( "{" +
+            "'dataElement':'%s'," +
+            "'categoryOptionCombo':'%s'," +
+            "'period':'202201'," +
+            "'orgUnit':'%s'," +
+            "'value':'24'," +
+            "'comment':'OK'}",
+            dataElementId, categoryOptionComboId, orgUnitId );
+
+        HttpResponse response = POST( "/dataValues", body );
```

Comment:
Just so you know you could have done the parameter replacement as part of the `POST`.
There is an overloaded version where you point out non parameter by wrapping them.
```java
POST( "/dataValues", Body(template), dataElementId, categoryOptionComboId, orgUnitId );
```
The `Body` here is a static method in `WebClient` that would be imported for better readability. 
The `template` would be the string for your body payload just that you put `{<name>}` placeholder in there.
The name here is just informal, the parameters are substituted in order given.

Reply:
Thanks, good to know.